src/MongoDB.Driver/ClusterKey.cs

```diff
@@ -195,5 +208,116 @@ public override int GetHashCode()
         {
             return _hashCode;
         }
+
+        // private methods
+        private void EnsureKmsProvidersAreValid()
```

Comment:
Remove this method.

See my branch.

Reply:
Oops... this comment was by mistake. We keep this method.

But, this does seem rather late to be ensuring that the kms providers are valid. Shouldn't we be doing that much earlier at the point where the application first set the value?